pkg/vip/util.go

```diff
@@ -104,6 +104,24 @@ func IsIPv6(address string) bool {
 	return ip.To4() == nil
 }
 
+func IsIPv4CIDR(cidr string) bool {
+	_, n, _ := net.ParseCIDR(cidr)
+	if n == nil {
+		return false
+	}
+	_, bits := n.Mask.Size()
+	return bits/8 == net.IPv4len
+}
+
+func IsIPv6CIDR(cidr string) bool {
+	_, n, _ := net.ParseCIDR(cidr)
+	if n == nil {
+		return false
+	}
+	_, bits := n.Mask.Size()
+	return bits/8 == net.IPv6len
+}
+
```

Comment:
Determining wheteher IP address is v4 or v6 is tricky, but I am not sure if length is always accurate to determine that. Maybe it would be better to just go with `net` package [IP.To4()](https://pkg.go.dev/net#IP.To4) and  [IP.To16()](https://pkg.go.dev/net#IP.To16)  functions?

Reply:
The `To4()` function is for IP addresses, not CIDR ranges which is the reason I had to add these additional two functions.